src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -83,7 +123,18 @@ public String beforeOrAfter(LocalDate someDate) {
      * @return Optional positive Integer
      */
     public Optional<Integer> diffBetweenZones(String firstZone, String secondZone) {
-        return Optional.empty();
+        ZonedDateTime dateFirst = null;
+        ZonedDateTime dateSecond = null;
+        LocalDateTime localDateTime = LocalDateTime.now();
+        Integer result = null;
+        try {
+            dateFirst = ZonedDateTime.of(localDateTime, ZoneId.of(firstZone));
+            dateSecond = ZonedDateTime.of(localDateTime, ZoneId.of(secondZone));
+            result = Math.abs((int)Duration.between(dateFirst,dateSecond).toHours());
```

Comment:
```suggestion
            result = Math.abs((int)Duration.between(dateFirst, dateSecond).toHours());
```

Reply:
Both lines are the same

Reply:
Did you mean to change 'result' for 'return'? I`ve changed it

Reply:
add white space after coma

Reply:
I`ve made it, thanks